Airside_Functions.cpp

```diff
@@ -0,0 +1,458 @@
+
+                            decoding_status = MAVLINK_DECODING_OKAY;                              
+                        }
+
+                        else if (warg_ID == MESSAGE_ID_GROUND_CMD)
+                        {                 
+                            PIGO_GROUND_COMMAND_t command;
+                            memset(&command, 0x00, sizeof(PIGO_GROUND_COMMAND_t));
+
+                            command.heading = global_position.lat;
+                            command.latestDistance = global_position.lon;
+
+                            memcpy((void*) telemetryData, (void*) &command, sizeof(PIGO_GROUND_COMMAND_t));
+
+                            decoding_status = MAVLINK_DECODING_OKAY;                               
+                        }
+
+                        else if (   warg_ID == MESSAGE_ID_WAYPOINT_MODIFY_PATH_CMD ||
+                                    warg_ID == MESSAGE_ID_WAYPOINT_NEXT_DIRECTIONS_CMD ||
+                                    warg_ID == MESSAGE_ID_HOLDING_TURN_DIRECTION)
+                        {
+                            one_byte_uint_cmd_t command;
+                            memset(&command, 0x00, sizeof(one_byte_uint_cmd_t));
+
+                            command.cmd = global_position.hdg;
+
+                            memcpy((void*) telemetryData, (void*) &command, sizeof(one_byte_uint_cmd_t));
+                            decoding_status = MAVLINK_DECODING_OKAY;
+                        }
+
+                        else if (   warg_ID == MESSAGE_ID_BEGIN_LANDING ||
+                                    warg_ID == MESSAGE_ID_BEGIN_TAKEOFF ||
+                                    warg_ID == MESSAGE_ID_INITIALIZING_HOMEBASE)
+                        {
+                            single_bool_cmd_t isLanded;
+                            memset(&isLanded, 0x00, sizeof(single_bool_cmd_t));
+
+                            isLanded.cmd = global_position.hdg;
+
+                            memcpy((void*) telemetryData, (void*) &isLanded, sizeof(single_bool_cmd_t));
+                            decoding_status = MAVLINK_DECODING_OKAY;
+                        }
+
+                        if (decoding_status == MAVLINK_DECODING_OKAY)
+                        {
+                            decoded_message_type = (PIGO_Message_IDs_e) warg_ID;
+                            *type = decoded_message_type;
+                            return MAVLINK_DECODING_OKAY;
+                        }
+
+                        return MAVLINK_DECODING_FAIL;
+                }
+            default:
+                break;
+        }// end of switch
+    }// if message receieved
+
+    return MAVLINK_DECODING_INCOMPLETE;
+}
+
+mavlink_encoding_status_t Mavlink_airside_encoder(POGI_Message_IDs_e msgID, mavlink_message_t *message, const uint8_t *struct_ptr) 
+{
+    mavlink_encoding_status_t encoding_status = MAVLINK_ENCODING_INCOMPLETE;
+
+    uint8_t system_id = 1;
+    uint8_t component_id = 1;
```

Comment:
are there enums for this? It will make it more understandable

Reply:
I set component_id to MAVLINK_COMM_0 since it's the default channel given by mavlink, the systemID can set to 1 cuz we are only controlling one drone